### ISSUE TITLE
Assigning labels/classes to unlabeled (-1) data in semi-supervised fit

### ISSUE THREAD

**Issue by ajf-ajf** [PAIR-code/umap-js#45]
Hi there,

Thanks for a wonderful library. Amazingly fast! Although I'm not quite sure how it's going to work for the set I'm using just yet.

Am I missing something, or is it just not possible to have labels assigned to previously unlabeled (-1) values? 

e.g., say I have a set of 100 labeled vectors, and a thousand or so that are unlabeled (that's not quite right, but it'll do). I've tried appending the thousand, using -1 for their labels, with `umap.setSupervisedProjection()` then `umap.fit()`. I've also tried running the above with just the labeled 100, then using `umap.transform()` with the extra 1000. Either way, everything gets projected into the reduced dimensional space, which is great. But, I don't see a way to extract which label umap identifies as most likely for unlabeled rows.

Short version is that I'm looking for a way to run a quick-n-not-that-dirty classification stage on some data prior to further analysis. And now I'm not sure whether this is something umap just doesn't do, or if I'm overlooking something super obvious, or if it's something that could be done through a step I'm also overlooking.

**Comment by ajf-ajf** [comment]
Oops, ok, so that was obvious:

https://umap-learn.readthedocs.io/en/latest/clustering.html
